opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/controller/PsuAuthController.java

```diff
@@ -0,0 +1,114 @@
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+import static de.adorsys.opba.restapi.shared.HttpHeaders.AUTHORIZATION_SESSION_ID;
+import static de.adorsys.opba.restapi.shared.HttpHeaders.X_REQUEST_ID;
+import static de.adorsys.opba.restapi.shared.HttpHeaders.X_XSRF_TOKEN;
+import static org.springframework.http.HttpHeaders.SET_COOKIE;
+
+@Slf4j
+@Controller
+@RequiredArgsConstructor
+public class PsuAuthController implements PsuAuthApi {
+    public static final Duration WEEK = Duration.ofDays(7);
+
+    private final PsuAuthService psuAuthService;
+    private final CookieProperties cookieProperties;
+
+    @Override
+    @SneakyThrows
+    public ResponseEntity<String> login(PsuAuthBody psuAuthBody, UUID xRequestID) {
+        Optional<Psu> psu = psuAuthService.getPsu(psuAuthBody.getId());
+        if (!psu.isPresent()) {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+
+        String jwtToken = generateToken(psu.get().getUserId(), getPrivateKey());
+
+        String sessionCookieString = ResponseCookie.from(AUTHORIZATION_SESSION_ID, jwtToken)
+                .httpOnly(cookieProperties.isHttpOnly())
+                .sameSite(cookieProperties.getSameSite())
+                .secure(cookieProperties.isSecure())
+                .path(cookieProperties.getPath())
+                .maxAge(cookieProperties.getMaxAge())
+                .build().toString();
+
+        return ResponseEntity
+                .status(HttpStatus.ACCEPTED)
+                .header(X_REQUEST_ID, xRequestID.toString())
+                .header(SET_COOKIE, sessionCookieString)
+                .header(X_XSRF_TOKEN, getHash(jwtToken))
+                .body(jwtToken);
+    }
+
+    @Override
+    public ResponseEntity<Void> registration(PsuAuthBody psuAuthDto, UUID xRequestID) {
+        Optional<Psu> psu = psuAuthService.createPsuIfNotExist(psuAuthDto.getId(), psuAuthDto.getPassword());
+        if (!psu.isPresent()) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+        HttpHeaders responseHeaders = new HttpHeaders();
+        responseHeaders.add(HttpHeaders.LOCATION, "/login");
+        return new ResponseEntity<>(responseHeaders, HttpStatus.CREATED);
+    }
+
+    @SneakyThrows
+    private String generateToken(String id, PrivateKey privateKey) {
+        ZonedDateTime currentTime = ZonedDateTime.now(ZoneOffset.UTC);
+        JWTClaimsSet claims = new JWTClaimsSet.Builder()
+                .expirationTime(Date.from(currentTime.plus(WEEK).toInstant()))
+                .issueTime(Date.from(currentTime.toInstant()))
+                .subject(String.valueOf(id))
+                .build();
+        JWSHeader jwsHeader = new JWSHeader.Builder(JWSAlgorithm.RS256).build();
+        SignedJWT signedJWT = new SignedJWT(jwsHeader, claims);
+        signedJWT.sign(new RSASSASigner(privateKey));
```

Comment:
RSASSASigner should be configurable bean

Reply:
done